packages/ts/form/src/Models.ts

```diff
@@ -34,57 +21,38 @@ export function hasFromString<T>(model: AbstractModel<T>): model is AbstractMode
   return _fromString in model;
 }
 
-export interface HasValue<T> {
-  value?: T;
-}
+export type Value<M> = M extends AbstractModel<infer T> ? T : never;
+
+export const _modelDetachedParent = { $value: undefined };
+
+export type ModelParent = AbstractModel | BinderNode | typeof _modelDetachedParent;
 
-export type ModelParent<T> = AbstractModel<any> | HasValue<T>;
-export type ModelValue<M extends AbstractModel<unknown>> = ReturnType<M['valueOf']>;
+export type EmptyModelConstructor<M> = new (
```

Comment:
If everything else is named as `detached model`, let's be consistent here as well.
```suggestion
export type DetachedModelConstructor<M> = new (
```